### PR TITLE
support clipping and trimming with remote storage

### PR DESCRIPTION
**Pull request by david-winder-kaltura**
PLAT-5667 #time 7d

### CODE REVIEW

File: alpha/apps/kaltura/lib/batch2/kFlowHelper.php [hunk 1]
```diff
@@ -251,6 +256,24 @@ public static function handleImportFinished(BatchJob $dbBatchJob, kImportJobData
 		return $dbBatchJob;
 	}
 
+	private function excuteClip($flavorAsset, $dbBatchJob) {
+		$entryFlavors = assetPeer::retrieveFlavorsByEntryId($flavorAsset->getEntryId());
```
Comment (MosheMaorKaltura) [inline alpha/apps/kaltura/lib/batch2/kFlowHelper.php]: rename to entryFlavorAssets

File: alpha/apps/kaltura/lib/batch2/kFlowHelper.php [hunk 2]
```diff
@@ -251,6 +256,24 @@ public static function handleImportFinished(BatchJob $dbBatchJob, kImportJobData
...
+		$originalFlavorAsset = assetPeer::retrieveOriginalByEntryId($flavorAsset->getEntryId());
+		foreach($entryFlavors as $entryFlavor) {
+			$flavor = assetParamsOutputPeer::retrieveByAsset($entryFlavor);
```
Comment (MosheMaorKaltura) [inline alpha/apps/kaltura/lib/batch2/kFlowHelper.php]: rename flavor  to flavorParamOutput

File: alpha/apps/kaltura/lib/batch2/kFlowHelper.php [hunk 3]
```diff
@@ -251,6 +256,24 @@ public static function handleImportFinished(BatchJob $dbBatchJob, kImportJobData
...
+			kBusinessPreConvertDL::decideFlavorConvert($entryFlavor, $flavor, $originalFlavorAsset, null, null, $dbBatchJob);
+		}
+	}
+	
+	private function isClipNeeded($dbEntry) {
```
Comment (MosheMaorKaltura) [inline alpha/apps/kaltura/lib/batch2/kFlowHelper.php]: Function should be static

File: alpha/apps/kaltura/lib/batch2/kFlowHelper.php [hunk 4]
```diff
@@ -251,6 +256,24 @@ public static function handleImportFinished(BatchJob $dbBatchJob, kImportJobData
 		return $dbBatchJob;
 	}
 
+	private function excuteClip($flavorAsset, $dbBatchJob) {
+		$entryFlavors = assetPeer::retrieveFlavorsByEntryId($flavorAsset->getEntryId());
+		$originalFlavorAsset = assetPeer::retrieveOriginalByEntryId($flavorAsset->getEntryId());
+		foreach($entryFlavors as $entryFlavor) {
+			$flavor = assetParamsOutputPeer::retrieveByAsset($entryFlavor);
+			kBusinessPreConvertDL::decideFlavorConvert($entryFlavor, $flavor, $originalFlavorAsset, null, null, $dbBatchJob);
```
Comment (MosheMaorKaltura) [inline alpha/apps/kaltura/lib/batch2/kFlowHelper.php]: Avoid using decideFlavorConvert. 
Do the following - 1 . Get source info. 2. Add Convert job on source.

File: alpha/apps/kaltura/lib/batch2/kFlowHelper.php [hunk 5]
```diff
@@ -251,6 +256,24 @@ public static function handleImportFinished(BatchJob $dbBatchJob, kImportJobData
 		return $dbBatchJob;
 	}
 
+	private function excuteClip($flavorAsset, $dbBatchJob) {
```
Comment (MosheMaorKaltura) [inline alpha/apps/kaltura/lib/batch2/kFlowHelper.php]: rename function to public static kBusinessPreConvertDL::convertSource($flavorAsset, $dbParentBatchJob=null)

File: alpha/apps/kaltura/lib/batch2/kFlowManager.php [hunk 6]
```diff
@@ -493,25 +493,30 @@ public function objectAdded(BaseObject $object, BatchJob $raisedJob = null)
 			{
 				if ($entry->getType() == entryType::MEDIA_CLIP)
 				{
-					$syncKey = $object->getSyncKey(flavorAsset::FILE_SYNC_FLAVOR_ASSET_SUB_TYPE_ASSET);
-
-					if (kFileSyncUtils::fileSync_exists($syncKey))
+					if ($entry->getOperationAttributes() && $object->getIsOriginal())
```
Comment (tan-tan-kanarek) [inline alpha/apps/kaltura/lib/batch2/kFlowManager.php]: Please use $entry->getDynamicFlavorAttributesForAssetParams instead of $entry->getOperationAttributes, only operation attributes on the source should matter.

File: api_v3/lib/KalturaEntryService.php [hunk 7]
```diff
@@ -685,9 +685,11 @@ protected function attachOperationResource(kOperationResource $resource, entry $
...
 		
 		$errDescription = '';
-		kBusinessPreConvertDL::decideAddEntryFlavor(null, $dbEntry->getId(), $resource->getAssetParamsId(), $errDescription, $dbAsset->getId(), $operationAttributes);
-		
-		if($isNewAsset)
+		$batchJob = kBusinessPreConvertDL::decideAddEntryFlavor(null, $dbEntry->getId(), $resource->getAssetParamsId(), $errDescription, $dbAsset->getId(), $operationAttributes);
+		$isImportNeeded = false;
+		if ($batchJob && $batchJob->getJobType() == BatchJobType::IMPORT)
+			$isImportNeeded = true;
+		if($isNewAsset && !$isImportNeeded)
```
Comment (tan-tan-kanarek) [inline api_v3/lib/KalturaEntryService.php]: Please check if you can check the $dbAsset status instead of the job type.

File: api_v3/lib/types/resource/KalturaEntryResource.php [hunk 8]
```diff
@@ -51,15 +51,12 @@ public function validateForUsage($sourceObject, $propertiesToSkip = array())
 		
 		$key = $srcFlavorAsset->getSyncKey(asset::FILE_SYNC_ASSET_SUB_TYPE_ASSET);
 		$c = FileSyncPeer::getCriteriaForFileSyncKey($key);
-		$c->addAnd(FileSyncPeer::FILE_TYPE, array(FileSync::FILE_SYNC_FILE_TYPE_FILE, FileSync::FILE_SYNC_FILE_TYPE_LINK), Criteria::IN);
+		$c->addAnd(FileSyncPeer::FILE_TYPE, array(FileSync::FILE_SYNC_FILE_TYPE_FILE, FileSync::FILE_SYNC_FILE_TYPE_URL), Criteria::IN);
```
Comment (tan-tan-kanarek) [inline api_v3/lib/types/resource/KalturaEntryResource.php]: Why did you remove links?
I think you should remove this condition all together and enable links.

File: api_v3/lib/KalturaEntryService.php (hunk 7 above)
Comment (MosheMaorKaltura) [inline api_v3/lib/KalturaEntryService.php]: Since the decision is done inside decideAddEntryFlavor, it more sense to check it return value than to add another logic.

File: api_v3/lib/types/resource/KalturaEntryResource.php [hunk 9]
```diff
@@ -51,15 +51,11 @@ public function validateForUsage($sourceObject, $propertiesToSkip = array())
 		
 		$key = $srcFlavorAsset->getSyncKey(asset::FILE_SYNC_ASSET_SUB_TYPE_ASSET);
 		$c = FileSyncPeer::getCriteriaForFileSyncKey($key);
-		$c->addAnd(FileSyncPeer::FILE_TYPE, array(FileSync::FILE_SYNC_FILE_TYPE_FILE, FileSync::FILE_SYNC_FILE_TYPE_LINK), Criteria::IN);
 		 
 		$fileSyncs = FileSyncPeer::doSelect($c);
-		foreach($fileSyncs as $fileSync)
-		{
-			$fileSync = kFileSyncUtils::resolve($fileSync);
-			if($fileSync->getFileType() == FileSync::FILE_SYNC_FILE_TYPE_FILE)
```
Comment (tan-tan-kanarek) [inline api_v3/lib/types/resource/KalturaEntryResource.php]: This test is still needed, check that the file-type is not link after the resolve.
